recipes/abseil/all/conanfile.py

```diff
@@ -78,32 +78,41 @@ def build(self):
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
             tools.patch(**patch)
         cmake = self._configure_cmake()
-        abi_h = tools.load("abi.h")
-        abi = dict()
-        for line in abi_h.splitlines():
-            if line.startswith("#define"):
-                tokens = line.split()
-                if len(tokens) == 3:
-                    abi[tokens[1]] = tokens[2]
-        for name, value in abi.items():
-            tools.replace_in_file(os.path.join(self._source_subfolder, "absl", "base", "options.h"),
-                    "#define ABSL_OPTION_{} 2".format(name),
-                    "#define ABSL_OPTION_{} {}".format(name, value))
+        abi_file = _ABIFile("abi.h")
+        abi_file.replace_in_options_file(os.path.join(self._source_subfolder, "absl", "base", "options.h"))
         cmake.build()
 
+    @property
+    def _module_path(self):
+        return os.path.join("lib", "cmake", "conan_trick")
+
+    @property
+    def _cxx_std_build_module(self):
+        return "cxx_std.cmake"
+
     def package(self):
         self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
         cmake = self._configure_cmake()
         cmake.install()
         tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
         cmake_folder = os.path.join(self.package_folder, "lib", "cmake")
-        self._create_components_file_from_cmake_target_file(os.path.join(cmake_folder, "absl", "abslTargets.cmake"))
+        components = self._create_components_file_from_cmake_target_file(os.path.join(cmake_folder, "absl", "abslTargets.cmake"))
         tools.rmdir(cmake_folder)
 
+        # Create a build-module that will propagate the required cxx_std to consumers of this recipe's targets
+        # TODO: Revisit with feedback from https://github.com/conan-io/conan/issues/10281
+        os.makedirs(os.path.join(self.package_folder, self._module_path))
+        with open(os.path.join(self.package_folder, self._module_path, self._cxx_std_build_module), 'w', encoding='utf-8') as f:
+            f.write("cmake_minimum_required(VERSION 3.1)\n\n")
```

Comment:
why?

Reply:
I think it's fine, as we use 3.15 anyway

Reply:
`target_compiler_features` were introduced in CMake 3.1, so at least use something that recognizes the function.